src/hll.c

```diff
@@ -248,6 +248,9 @@ HyperLogLog_reduce(HyperLogLog *self)
 
     PyObject *args = Py_BuildValue("(ii)", self->k, self->seed);
     PyObject *registers = Py_BuildValue("s#", arr, self->size);
+
+    free(arr);
```

Comment:
To fix all the leaks both `args` and `registers` need to have their reference counts manually decremented since `PyBuildValue` creates its own copies:

```suggestion
    free(arr);

    PyObject *result = Py_BuildValue("(OOO)", Py_TYPE(self), args, registers);
    Py_DECREF(args);
    Py_DECREF(registers);
    return result;
```

Reply:
Instead of manually decreasing the reference counts, `Py_BuildValue` offers a handy `N` format argument that avoids incrementing the reference counts in the first place.

> "N" (object) [PyObject *]
Same as "O", except it doesn't increment the reference count on the object. Useful when the object is created by a call to an object constructor in the argument list.

Reply:
Using the example from #33:

```python
... from HLL import HyperLogLog
... import pickle
... import os
... import psutil
... 
... process = psutil.Process(os.getpid())
... def print_memory_usage():
...     B = process.memory_info().rss
...     MB = B / (1<<20)
...     print(f'{MB:.5f} MB')
... 
... 
... for i in range(10000):
...     hll = HyperLogLog(14)
...     pickle.dumps(hll)
...     if i % 1000 == 0:
...         print_memory_usage()
...
28.00781 MB
28.00781 MB
28.00781 MB
28.00781 MB
28.00781 MB
28.00781 MB
28.00781 MB
28.00781 MB
28.00781 MB
28.00781 MB

>> hll1 = HyperLogLog(16)
>> hll1.add("hi!")
>> hll1.cardinality()
1.0000076294721387
>> dump = pickle.dumps(hll1)
>> hll2 = pickle.loads(dump)
>> hll2.cardinality()
1.0000076294721387
```